### DISCUSSION TITLE
How to pass parameters to child in nested query?

### DISCUSSION THREAD

**Discussion by CuBeof** [mirumee/ariadne#573] [Q&A]

**Comment by pylipp** [comment]
I assume you have a scenario like 

```
type Book {
  id: ID!
  title: String
}
type Author {
  id: ID!
  name: String
  books: [Book!]!
}
type Query {
  author(id: ID!): Author
}
```

In order to make queries like
```
query { author(id: 42) { books(nr: 10) { title } } }
```
to return the first 10 book titles of author identified by ID 42, update the schema for the author type: `books(nr: Int): [Book!]!`

The corresponding resolver then is along the lines of
```
@author.field("books")
def resolve_author_books(author_obj, info, nr):
    # call to retrieve nr rows of data
```

This is mentioned in the GraphQL [base documentation](https://graphql.org/learn/queries/#arguments). I also found [this answer](https://stackoverflow.com/a/61493044/3865876) helpful.

Hope that helps =)
